src/opnsense/mvc/app/models/OPNsense/Base/FieldTypes/FilterProtocolField.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ *    Copyright (C) 2015-2019 Deciso B.V.
+ *    Copyright (C) 2020 Fabian Franz
+ *    Copyright (C) 2004-2007 Scott Ullrich <sullrich@gmail.com>
+ *    Copyright (C) 2005 Bill Marquette <bill.marquette@gmail.com>
+ *    Copyright (C) 2006 Peter Allgeyer <allgeyer@web.de>
+ *    Copyright (C) 2008-2010 Ermal Luçi
+ *    Copyright (C) 2003-2004 Manuel Kasper <mk@neon1.net>
```

Comment:
Personally, this feels like a wasted opportunity to cleanly rebuild the code. Already nobody knows what these copyrights were for in terms of code and it's historical dead weight for going forward.

Reply:
I copied a single function and I cannot identify who is the originator so I have to take all of them with it…

Reply:
So you don't copy the function.

Reply:
I just need a variant of getProtocols(), which is from here: get_protocols from src/etc/inc/filter.inc

I need a hash structure instead of an array so I have to convert that anyway (means it is not totally identical to the original).
The original also has some issues with up/downcase handling because the legacy code seems to handle the return value differently.